llvm/lib/ExecutionEngine/JITLink/ELF_aarch32.cpp

```diff
@@ -200,6 +200,9 @@ class ELFLinkGraphBuilder_aarch32
 
 protected:
   TargetFlagsType makeTargetFlags(const typename ELFT::Sym &Sym) override {
+    // Data symbols do not have Arm or Thumb flags.
+    if (Sym.getType() == ELF::STT_OBJECT)
+      return TargetFlagsType{};
```

Comment:
With your fix in `getRawOffset()` this check shouldn't be necessary anymore, isn't it? Instead we might want to assert that all symbols with the `ThumbSymbol` target flag are `ELF::STT_FUNC`.

Reply:
Agree but I had to keep it because of a failed assert in https://github.com/llvm/llvm-project/blob/f1faba25433c971f024dd8a29da14020246e89ec/llvm/lib/ExecutionEngine/Orc/ObjectLinkingLayer.cpp#L52

Reply:
> With your fix in `getRawOffset()` this check shouldn't be necessary anymore, isn't it? Instead we might want to assert that all symbols with the `ThumbSymbol` target flag are `ELF::STT_FUNC`.

Updated the condition to check for `STT_FUNC`. I found a similar assert to what you mentioned in the previous comment, should I still add one?